lib/stdlib/src/binary.erl

```diff
@@ -934,6 +935,29 @@ unhex(X) ->
              no, 10, 11, 12, 13, 14, 15, no, no, no, no, no, no, no, no, no  %96
             }).
 
+-doc """
+Joins a list of binaries together by a specified Separator.
+Equivalent to `iolist_to_binary(lists:join(Separator, Binaries))` with binary:join/2 being the faster of the two.
+
+_Example:_
+
+```erlang
+1> binary:join([<<"a">>, <<"b">>, <<"c">>], <<", ">>).
+<<"a, b, c">>
+```
+""".
+-doc(#{since => <<"OTP 27.0">>}).
+-spec join([binary()], binary()) -> binary().
+join([], _Separator) -> <<>>;
+join([H], _Separator) -> H;
+join([H | T], Separator) ->
+    join(T, Separator, H).
```

Comment:
This can be improved to convince the compiler to use the new "private append" optimisation by making sure we start appending from an empty binary.
In my very simple & naive benchmarks this makes this ~20% faster on the data I used - it likely needs to be tested more thoroughly 
```suggestion
    Acc = <<>>,
    join(T, Separator, <<Acc/binary, H/binary>>).
```

Reply:
Getting some mixed results on my benchmark (it's also a very naive benchmark 😄) but it does indeed consistently perform better with anywhere between 20-50% improvement 😮 👍

Reply:
Might be nice to add a comment in the code explaining why this is needed.